### PR TITLE
Add zlib submodule and modified cmake file

### PR DESCRIPTION
**Pull request by yhyu13**
Solve #69 where zlib is added as a submodule

### CODE REVIEW

**Comment by yhyu13** [comment]
zlib.dll that is built is thus required on shadred library build

File: CMakeLists.txt [hunk 1]
```diff
@@ -117,7 +159,6 @@ endif()
 
 set(USE_ZLIB ON)
 add_definitions( -DUSE_ZLIB )
```
Comment (Lecrapouille) [inline CMakeLists.txt:161]: You have removed add_definitions( -DHAVE_AES ) ? Is this ok ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -161,7 +195,7 @@ if(MSVC OR USING_INTEL)
             if(${flag_var} MATCHES "/MD")
                 string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
             endif(${flag_var} MATCHES "/MD")
-
+
```
Comment (Lecrapouille) [inline CMakeLists.txt:198]: Can you delete trailing white spaces, all along the file ?

**Comment by Lecrapouille** [comment]
@yhyu13 Thanks I'll test it !

File: CMakeLists.txt [hunk 3]
```diff
@@ -106,8 +106,50 @@ if("${CMAKE_BUILD_TYPE}" STREQUAL "")
 or CMAKE_C_FLAGS are used), Debug, Release, RelWithDebInfo, MinSizeRel." FORCE)
 endif()
 
-find_package(ZLIB REQUIRED)
+# Foce use zlib submodule
+#find_package(ZLIB REQUIRED)
+if(TRUE)
```
Comment (Lecrapouille) [inline CMakeLists.txt:111]: Could not be better ?

**Review by Lecrapouille — COMMENTED** [review]
Afaik, I cannot compile this patch I have this error on Debian 10:
```
[ 11%] Built target ZLIB_EXTERNAL
[ 39%] Built target staticZipper
make[2]: ***  Aucune règle pour fabriquer la cible « zlib_install/lib/zlib.lib », nécessaire pour « Zipper-test ». Arrêt.
make[1]: *** [CMakeFiles/Makefile2:115: CMakeFiles/Zipper-test.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

**Comment by yhyu13** [comment]
@Lecrapouille I've updated the cmake file such that only windows uses the zlib submodule

File: CMakeLists.txt (hunk 1 above)
Comment (yhyu13) [reply CMakeLists.txt]: added back

**Comment by Lecrapouille** [comment]
@yhyu13  Sorry this still failing https://travis-ci.org/github/sebastiandev/zipper/jobs/762553598
Sorry I dunno why Travis-Ci no longer shows the status. I'm a little bored of these CI always broken by their maintainers :(

Else concerning your patch CMakeLists.txt I'm not found of `if(WIN32)` this could be used for Linux too. Ideally, I would have preferred to see the as `OPTION(USE_NESTING_ZLIB "Use the zlib included as submodule", OFF)`. If `OFF` you can force it to `ON` and call your new code if `if(NOT EXISTS "${ZLIB_INCLUDE_DIRS}/zlib.h")` is triggered.

What about the zlib.h inside the submodule: is it installed to the `/usr/include` when you do `sudo make install` ? Does not conflict with the original zlib installed with `apt-get install` ?

File: README.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ In order to use and compile zipper you need to have [zlib](http://www.zlib.net)
 **Zipper** depends on minizip as well but since it is used as a submodule, you get it when cloning
 the repo and it gets compiled with the project.
 
-*Note*: For windows users, zlib is expected to be found at ZLIBROOT.
+*Note*: For windows user, zlib is contained as a submodule. Due to how zlib named its libaray, you would need to build the whole project with release first, and then copy the zlib.dll under ./zlib_bin/Release to your executable.
```
Comment (Lecrapouille) [inline README.md:28]: libaray => typo

**Comment by jnhyatt** [comment]
Have you considered adding this as an external project instead? If targets are exported correctly, you could add this using CMake ExternalProject, then set it to install to a directory within the build directory. From there, you simply add that install directory to the CMake prefix path and call find_package. This doesn't work with the main zlib repository, but Mizux has a fork that has properly exported modern CMake targets. The reason I'd prefer that approach is because it follows the more modern CMake paradigm, so the user of this library could decide whether they want to use their own installed zlib version or the zlib version included by the library. So you could add an option to add the external project, and if you don't, it'll just check the user's install directory. Thoughts?

**Comment by Lecrapouille** [comment]
I close this PR since #93 was commited first. Feel free to reopen it if needed

**Pull request closed without merging**